src/app/public/modules/progress-indicator/progress-indicator.component.spec.ts

```diff
@@ -20,387 +20,649 @@ import {
-      fixture.detectChanges();
-      tick();
+    it('should hide the next button and show the finish button on the last step', fakeAsync(function () {
+      detectChanges();
 
-      expect(fixture.nativeElement.querySelector('#previous-btn button').innerText.trim()).toBe('Previous');
-      expect(fixture.nativeElement.querySelector('#next-btn button').innerText.trim()).toBe('Next');
+      let nextButtonElement = getNavButtonElement('next');
+      let finishButtonElement = getNavButtonElement('finish');
+
+      expect(nextButtonElement).toBeTruthy();
+      expect(finishButtonElement).toBeFalsy();
+
+      gotoStep(2);
+      detectChanges();
+
+      nextButtonElement = getNavButtonElement('next');
+      finishButtonElement = getNavButtonElement('finish');
+
+      expect(nextButtonElement).toBeFalsy();
+      expect(finishButtonElement).toBeTruthy();
     }));
 
-    it('should use inputted disabled state', fakeAsync(() => {
-      fixture.componentInstance.previousButtonDisabled = true;
-      fixture.componentInstance.nextButtonDisabled = true;
-      fixture.componentInstance.resetButtonDisabled = true;
-      fixture.detectChanges();
-      tick();
-      fixture.detectChanges();
-      let buttonElements = fixture.nativeElement.querySelectorAll('sky-progress-indicator-nav-button button');
-      let resetButton = fixture.nativeElement.querySelector('#reset-btn button');
+    it('should not hide the next button if the finish button does not exist', fakeAsync(function () {
+      // Create a custom button config that does not include a finish button.
+      componentInstance.buttonConfigs = [
+        {
+          type: 'previous'
+        },
+        {
+          type: 'next'
+        }
+      ];
 
-      expect(fixture.componentInstance.navButtons.first.disabled).toBeTruthy();
-      expect(buttonElements[0].disabled).toBeTruthy();
+      detectChanges();
 
-      expect(fixture.componentInstance.navButtons.last.disabled).toBeTruthy();
-      expect(buttonElements[1].disabled).toBeTruthy();
+      let nextButtonElement = getNavButtonElement('next');
 
-      expect(fixture.componentInstance.resetButton.disabled).toBeTruthy();
-      expect(resetButton.disabled).toBeTruthy();
+      expect(nextButtonElement).toBeTruthy();
+
+      gotoStep(2);
+      detectChanges();
+
+      nextButtonElement = getNavButtonElement('next');
+
+      expect(nextButtonElement).toBeTruthy();
+      expect(nextButtonElement.disabled).toBeTruthy();
     }));
 
-    it('should be accessible', async(() => {
+    it('should throw error if progress indicator not set as an input', fakeAsync(function () {
+      componentInstance.defaultNavButtonProgressIndicatorRef = undefined;
+
+      try {
+        detectChanges();
+        fail('It should throw error!');
+      } catch (error) {
+        expect(error).toExist();
+      }
+    }));
+  });
+
+  describe('Accessibility', function () {
+    it('should be accessible', async(function () {
```

Comment:
We sure should! I'll make that change.

Reply:
There's some different DOM elements in horizontal mode - should we have an accessibility test for that too?